### PR TITLE
BUG: Fix .dt.microsecond accessor for pyarrow-backed Series

### LINKED ISSUE #59154: BUG: `.dt.microsecond` for pyarrow-backed Series returns 0
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
In [22]: s = pd.Series(pd.date_range('2000', periods=3, freq='15ms'))

In [23]: s.dt.microsecond
Out[23]:
0        0
1    15000
2    30000
dtype: int32

In [24]: s.convert_dtypes(dtype_backend='pyarrow')
Out[24]:
0           2000-01-01 00:00:00
1    2000-01-01 00:00:00.015000
2    2000-01-01 00:00:00.030000
dtype: timestamp[ns][pyarrow]

In [25]: s.convert_dtypes(dtype_backend='pyarrow').dt.microsecond
Out[25]:
0    0
1    0
2    0
dtype: int64[pyarrow]
```

### Issue Description

For non-pyarrow backed dtype, it returns the total number of microseconds since the last second

For pyarrow-backed, it just returns 0

### Expected Behavior
```python
In [25]: s.convert_dtypes(dtype_backend='pyarrow').dt.microsecond
Out[25]:
0        0
1    15000
2    30000
dtype: int64[pyarrow]
```

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit                : d9cdd2ee5a58015ef6f4d15c7226110c9aab8140
python                : 3.11.9.final.0
python-bits           : 64
OS                    : Linux
OS-release            : 5.15.153.1-microsoft-standard-WSL2
Version               : #1 SMP Fri Mar 29 23:14:13 UTC 2024
machine               : x86_64
processor             : x86_64
byteorder             : little
LC_ALL                : None
LANG                  : C.UTF-8
LOCALE                : en_US.UTF-8

pandas                : 2.2.2
numpy                 : 2.0.0
pytz                  : 2024.1
dateutil              : 2.9.0.post0
setuptools            : 65.5.0
pip                   : 24.0
Cython                : None
pytest                : 8.1.1
hypothesis            : 6.100.1
sphinx                : None
blosc                 : None
feather               : None
xlsxwriter            : None
lxml.etree            : None
html5lib              : None
pymysql               : None
psycopg2              : None
jinja2                : 3.1.3
IPython               : 8.23.0
pandas_datareader     : None
adbc-driver-postgresql: None
adbc-driver-sqlite    : None
bs4                   : 4.12.3
bottleneck            : None
dataframe-api-compat  : None
fastparquet           : None
fsspec                : 2024.3.1
gcsfs                 : None
matplotlib            : 3.8.4
numba                 : None
numexpr               : None
odfpy                 : None
openpyxl              : None
pandas_gbq            : None
pyarrow               : 16.1.0
pyreadstat            : None
python-calamine       : None
pyxlsb                : None
s3fs                  : None
scipy                 : 1.13.0
sqlalchemy            : None
tables                : None
tabulate              : None
xarray                : None
xlrd                  : None
zstandard             : None
tzdata                : 2024.1
qtpy                  : None
pyqt5                 : None

</details>

### PR DESCRIPTION
**Pull request by Aloqeely**
- [x] closes #59154(Replace xxxx with the GitHub issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: doc/source/whatsnew/v3.0.0.rst [hunk 1]
```diff
@@ -496,6 +496,7 @@ Datetimelike
 - Bug in :func:`date_range` where the last valid timestamp would sometimes not be produced (:issue:`56134`)
 - Bug in :func:`date_range` where using a negative frequency value would not include all points between the start and end values (:issue:`56382`)
 - Bug in :func:`tseries.api.guess_datetime_format` would fail to infer time format when "%Y" == "%H%M" (:issue:`57452`)
+- Bug in :math:`Series.dt.microsecond` producing incorrect results for pyarrow backed :class:`Series`. (:issue:`59154`)
```
Comment (mroeschke) [inline doc/source/whatsnew/v3.0.0.rst:499]:
```suggestion
- Bug in :meth:`Series.dt.microsecond` producing incorrect results for pyarrow backed :class:`Series`. (:issue:`59154`)
```

File: pandas/tests/extension/test_arrow.py [hunk 2]
```diff
@@ -2456,7 +2456,7 @@ def test_dt_properties(prop, expected):
                 hour=3,
                 minute=4,
                 second=7,
-                microsecond=5,
+                microsecond=2000,
```
Comment (mroeschke) [inline pandas/tests/extension/test_arrow.py:2459]: Could you also add a separate test where microseconds should return 0?

Comment (Aloqeely) [reply pandas/tests/extension/test_arrow.py:2459]: Sure

**Review by MarcoGorelli — APPROVED** [review]
thanks @Aloqeely !

**Review by mroeschke — APPROVED** [review]

**Pull request merged**